src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1251,20 +1251,29 @@ private Map<String, List<String>> loadVanityPaths(ResourceResolver resolver) {
         log.debug("end vanityPath query; elapsed {}ms", TimeUnit.NANOSECONDS.toMillis(queryElapsed));
 
         long count = 0;
+        long countInScope = 0;
         long processStart = System.nanoTime();
 
         while (i.hasNext()) {
             count += 1;
             final Resource resource = i.next();
             final String resourcePath = resource.getPath();
             if (Stream.of(this.factory.getObservationPaths()).anyMatch(path -> path.matches(resourcePath))) {
+                countInScope += 1;
                 final boolean addToCache = isAllVanityPathEntriesCached()
                         || vanityCounter.longValue() < this.factory.getMaxCachedVanityPathEntries();
                 loadVanityPath(resource, resolveMapsMap, targetPaths, addToCache);
             }
         }
         long processElapsed = System.nanoTime() - processStart;
-        log.debug("processed {} vanityPaths in {}ms", count, TimeUnit.NANOSECONDS.toMillis(processElapsed));
+        log.debug("processed {} vanityPaths (of which {} in scope) in {}ms", count, countInScope, TimeUnit.NANOSECONDS.toMillis(processElapsed));
+        if (!isAllVanityPathEntriesCached()) {
+            if (countInScope > this.factory.getMaxCachedVanityPathEntries()) {
+                log.warn("Number of vanity paths in scope ({}) exceeds configured cache size ({})", countInScope, this.factory.getMaxCachedVanityPathEntries());
```

Comment:
What is the consequence of this? 
Also can we add a call-to-action to this? Would "Increase the cache size" be the right recommendation?

Reply:
The consequence is that request for certainl paths will be slow because they'll hit the storage (by query). The recommendation would be to either reduce the number of vanity paths, or increase the cache size.

Do you want to make a proposal for the concrete text?

Reply:
```suggestion
                log.warn("Number of vanity paths in scope ({}) exceeds configured cache size ({}); this and any following vanityPath will be much slower. Consider increasing the cache size or decrease the number of vanity paths", countInScope, this.factory.getMaxCachedVanityPathEntries());
```